src/main/java/org/openrewrite/staticanalysis/NeedBraces.java

```diff
@@ -69,20 +69,42 @@ private static class NeedBracesVisitor extends JavaIsoVisitor<ExecutionContext>
          * We can use that to our advantage by saying if you aren't a block (e.g. a single {@link Statement}, etc.),
          * then we're going to make this into a block. That's how we'll get the code bodies surrounded in braces.
          */
-        private <T extends Statement> J.Block buildBlock(Statement owner, T element) {
-            J j = getCursor().getParentTreeCursor().getValue();
+        private <T extends Statement> J.Block buildBlock(T element) {
+            J rootElement = null;
             Space end = Space.EMPTY;
-            if (j instanceof J.Block) {
-                J.Block block = (J.Block) j;
+
+            Cursor currentCursor = getCursor();
+            while (
+                    currentCursor != null &&
+                            currentCursor.getParent() != null &&
+                            !(currentCursor.getParent().getValue() instanceof J.Block)
+            ) {
+                currentCursor = currentCursor.getParent();
+            }
+
+            if (currentCursor != null && currentCursor.getValue() instanceof JRightPadded) {
+                JRightPadded<J> paddedIf = currentCursor.getValue();
+                rootElement = paddedIf.getElement();
+            }
+
+            if (rootElement instanceof Statement) {
+                Cursor blockParentCursor = currentCursor.getParent();
+                J.Block block = blockParentCursor.getValue();
                 List<Statement> statements = block.getStatements();
-                int i = statements.indexOf(owner);
-                boolean last = i == statements.size() - 1;
-                Space trailingSpace = last ? block.getEnd() : statements.get(i + 1).getPrefix();
-                if (!trailingSpace.getComments().isEmpty() && trailingSpace.getWhitespace().indexOf('\n') == -1) {
-                    end = trailingSpace;
-                    getCursor().getParentTreeCursor().<List<Integer>>computeMessageIfAbsent("replaced", k -> new ArrayList<>()).add(i);
+                int currentIndex = statements.indexOf(rootElement);
+                boolean last = currentIndex == statements.size() - 1;
+                Space trailingComment = last ? block.getEnd() : statements.get(currentIndex + 1).getPrefix();
+
+                if (!trailingComment.isEmpty() && !trailingComment.getWhitespace().contains("\n")) {
+                    end = trailingComment;
+                    if (last) {
+                        blockParentCursor.putMessage("removeEndComments", true);
+                    } else {
+                        blockParentCursor.<List<Integer>>computeMessageIfAbsent("replaced", k -> new ArrayList<>()).add(currentIndex);
```

Comment:
messages can appear at the end of the block for cases such as.

```
static void method() {
    if (true) return; // if comment
    else return; // else comment
}
```

for this case the `// else comment` would be located at the `end` of the block